### PR TITLE
Effect tweaks

### PR DESCRIPTION
**Pull request by kaadmy**
Title.

Changes:

- Modify the RG's trail a bit
- Remove pubes on the LG's impact and add sparks instead
- Add sparks to the impact effects of the RG and HB
- Make blood and gib trails look more like blood
- Make explosions look meatier and denser
- Make rocket/grenade smoke trails more visible
- Change bullet tracers and impacts to be more realistic/beefier
- Tweak water splashes a little

### CODE REVIEW

File: src/cgame/default/cg_temp_entity.c [hunk 1]
```diff
@@ -595,92 +628,84 @@ static void Cg_RailEffect(const vec3_t start, const vec3_t end, const vec3_t dir
...
-	// draw the core with a beam
+	// Rail core
 
 	if (!(p = Cg_AllocParticle(PARTICLE_BEAM, cg_particles_beam))) {
 		return;
 	}
 
-	// white cores for some colors, shifted for others
-	// FIXME: what to do here for 24-bit colors?
-	ColorToVec4(ColorFromRGB(235, 235, 235), p->color_start);
```
Comment (jdolan) [inline src/cgame/default/cg_temp_entity.c]: Guess we answered that question! ;)

Comment (kaadmy) [inline src/cgame/default/cg_temp_entity.c]: LOL yeah :D
There were quite a few outdated comments in the effect code; should I remove them and try to standardize effects?
Almost all the code still uses color constants and palette colors, should they be replaced with `vec4`s etc in favor of palette colors?
The code changes how the same thing is handled in different places, for example, gravity on some particles uses `VectorSet(p->accel, 0.0, 0.0, -PARTICLE_GRAVITY);` while some others use `p->accel[2] = -PARTICLE_GRAVITY`

**Comment by kaadmy** [comment]
I tweaked underwater effects a bit more, make explosions have no sparks underwater and made bullets have bubbles instead of sparks+smoke.

**Comment by Panjoo** [comment]
Some FX look better but I also noticed some that aren't there yet. I'll take a closer look at these changes soon.

But one thing popped out and that's this:
I assume this is a temp particle texture because this just looks strange.
rgpartic

**Comment by kaadmy** [comment]
It's supposed to be like that, but I just made that in GIMP in about 10 seconds :D
I'm fine with you making a new one if you want to.

**Comment by Panjoo** [comment]
10 seconds... tsk tsk :-)
Alright I'll cook something up then.

**Comment by kaadmy** [comment]
Okay I kid, it was more like 20 seconds :P
(GIMP takes about 10 seconds to start up)
If you want I can modify the trail too to fit a specific particle image better.

**Pull request merged**